datajunction-query/djqs/engine.py

```diff
@@ -66,22 +71,61 @@ def get_columns_from_description(
     return columns
 
 
+def create_trino_engine(engine: Engine, headers: Optional[Dict[str, str]]) -> "Engine":
+    """
+    Create a SQLAlchemy engine for Trino.
+    """
+    # Parsing the URI
+    parsed_uri = urlparse(engine.uri)
+    scheme = parsed_uri.scheme
+    host = parsed_uri.hostname
+    port = parsed_uri.port
+    path = parsed_uri.path.lstrip("/")
+    catalog, schema = path.split("/", 1) if "/" in path else (path, None)
+
+    global QUERY_USER, QUERY_PASSWORD, HTTP_SCHEME
+
+    # Override user and password from headers if available
+    if headers and "QUERY_USER" in headers and "QUERY_PASSWORD" in headers:
```

Comment:
Instead of doing this, you can just do `query_user = headers.get("QUERY_USER")` and `query_password = headers.get("QUERY_PASSWORD")` and then later when you use them you can check if they're `None`.
```py
    if HTTP_SCHEME == "https" and query_user and query_password:
        engine_str = f"{scheme}://{QUERY_USER}:{QUERY_PASSWORD}@{host}:{port}/{catalog}/{schema}?protocol={HTTP_SCHEME}"
        return create_engine(engine_str)
    else:
        engine_str = engine.uri
        return create_engine(engine_str, connect_args=engine.extra_params)
```